dataline-server/src/main/java/io/dataline/server/ServerApp.java

```diff
@@ -59,6 +69,12 @@ public ServerApp(final Path configRoot) {
   }
 
   public void start() throws Exception {
+    // hack: upon installation we need to assign a random customerId so that when
+    // tracking we can associate all action with the correct anonymous id.
+    setCustomerIdIfNotSet(configRoot);
```

Comment:
This is really not great. I like it better than the other option I could come up with though. The alternative I think is to have an initialize script in our docker setup that finds the default workspace json file and mutates it to inject a customerId. I don't like it as much as what I have here for two reasons:
1. Don't want to add initialize scripts in docker right now since we don't do this for anything else.  
1. It's a hack but at least it's pretty easy to find and understand when it's in the java code as opposed to some random bash script.